### PR TITLE
Support for configuring dex theme via values.yaml

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
An important requirement from our customers is to add their own branding for our dashboard and dex. For updating the theme of dex, users can update the files in [theme](https://github.com/kubermatic/kubermatic/tree/release/v2.23/charts/oauth/theme) directory which are then propagated and installed in the form of a [kubernetes secret](https://github.com/kubermatic/kubermatic/blob/release/v2.23/charts/oauth/templates/secrets.yaml#L32).

While this works it also adds a huge limitation since all the files need to exist and be up-to-date in the `theme` directory before running the installer. This PR adds a way to consume those assets from `values.yaml` as well. With this change, we don't need to maintain those files anymore, all the user needs is to maintain a single values file.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Support for configuring the dex theme via values file.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: charts/oauth/values.yaml [hunk 1]
```diff
@@ -106,3 +106,8 @@ dex:
...
...admeaeG1Xx7g1x/ZxOF9j7xJX7DGYnJLnqbkSYKJ2rhmQtZo47ynaSytEFCV2kBZl5/sCgDgTnN6vNBhfaJC89njx3h01y7+8NhjSy2whSXPDXk7IY0i4oUWyUKLqBWjVaBdFETW0hDB6crIpt+cHesKAOBT7bNTz+565GCjtw8nwqEf71t0IGCKgiRLacUJSSsib7SIGk2cyyniBNtKsGmGMQUZoIK+78L6l/0k+9jh5nDLqbHyl/bSs/cLvP7KK4sOBNKiwCKUvKHPGHoqVeJKhRXLq5gooeI8Je9QpgrVCuLkomG8rAOCjIkI8qFbMTdtxgNHnzmEiGCsIc0zkjRF0jZFGmODI43amCghRAm0EnScUYozcBdvxst/EaWWqg3UcotsugcnwoM7dnD8hRc76zgEsA4pCuadJy8MeTvFxm1C3IY4RUcpKsnBuu4BWlE6HeYjQqONL/VjrrmWqNXk0T17QKSzEZ0jco7ceUKe47IMl6bYdoZvZ4R2jhQW7/x01wAvjQ2PTy+ca87OztM4t0Dj3etxIgQJHXHvEefInEeMgaLA5zmmMGR5QZJb4tzStg5j3ZGuhxCgGSd3pXE2UU/bSP8wfXTe+4h0AEIACZ31HAJeKbJSCafLlEoWVXH0E1De7e5+BoDZ744eacbx9kac0KitxpZreASCB+/f1AqL8hbnLbnzJM7TdJ5EBTJn7//rvTd034Lzkfz87vE4aq3P0nRqfvAGROiIBllqBcETQsCLYEQoAFsGXQ7jJ7/54e9fqm7pSgEA/OuHz7h//e6n2XWfOFEt4pFW//sGCL5jvwK0hnIFqjXo7YFlNajp3cXeO+7/bzW7Ajgf7syfp1prvzyIhFElHg2okkYqZajXYVkvLKtNUy3fxb7N429Vq6uf0wtDi1AClFZIuQK9PYTe2hRV/TD7Pzl+JTXeFkBNQR3VDJXKlKnXnpJK5WB48p7pbmr8B3ATxATU5NDqAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDIzLTAyLTIzVDE0OjIxOjEyKzAwOjAwDYmh3gAAACV0RVh0ZGF0ZTptb2RpZnkAMjAyMy0wMi0yM1QxNDoyMToxMiswMDowMHzUGWIAAAAodEVYdGRhdGU6dGltZXN0YW1wADIwMjMtMDItMjNUMTQ6MjE6MTIrMDA6MDArwTi9AAAAAElFTkSuQmCC
```
Comment (ahmedwaleedmalik) [inline charts/oauth/values.yaml:113]: Just wanted to keep it here as an example for the users.

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**